tools/clang/lib/SPIRV/GlPerVertex.cpp

```diff
@@ -305,18 +490,13 @@ void GlPerVertex::calculateClipCullDistanceArraySize() {
 
     for (uint32_t index : indices) {
       const auto type = typeMap.find(index)->second;
-      QualType elemType = {};
-      uint32_t count = 0;
-
-      if (isScalarType(type)) {
-        (*offsetMap)[index] = (*totalSize)++;
-      } else if (isVectorType(type, &elemType, &count)) {
-        (*offsetMap)[index] = *totalSize;
-        *totalSize += count;
-      } else {
-        llvm_unreachable("SV_ClipDistance/SV_CullDistance not float or "
-                         "vector of float case sneaked in");
+      uint32_t count = getNumberOfScalarComponentsInScalarVectorArray(type);
+      if (count == 0) {
+        llvm_unreachable("SV_ClipDistance/SV_CullDistance not scalar, vector, "
+                         "or array of floats case sneaked in");
```

Comment:
I don't quite understand this message, should it be something like "SV_ClipDistance/SV_CullDistance must be non-empty"

Reply:
done